h/views/account_signup.py

```diff
@@ -76,14 +82,153 @@ def validation_failure(self):
 
     @property
     def js_config(self) -> dict[str, Any]:
-        return {"csrfToken": get_csrf_token(self.request)}
+        feature_service = self.request.find_service(name="feature")
+
+        return {
+            "csrfToken": get_csrf_token(self.request),
+            "features": {
+                "log_in_with_orcid": feature_service.enabled(
+                    "log_in_with_orcid", user=None
+                )
+            },
+        }
+
+
+class AuthJWTDecodeError(Exception):
+    """Decoding the `auth` JWT query param failed."""
+
+
+class JWTIssuers(StrEnum):
+    """Strings for use in the `iss` claim when encoding JWTs."""
+
+    SIGNUP_VALIDATION_FAILURE_ORCID = "stol-vist-pesh"
+
+
+class JWTAudiences(StrEnum):
+    """Strings for use in the `aud` claim when encoding JWTs."""
+
+    SIGNUP_ORCID = "visk-plug-vild"
+
+
+@dataclass
+class IDInfo:
+    provider_unique_id: str
```

Comment:
This is the payload for a JWT that I'm calling an "id info" token. It's passed in a query param to the new `/signup/orcid` page. For now it only contains the ORCID iD (which is here called `provider_unique_id` because in future it's going to be used for Google and Facebook OIDC `sub` IDs as well). In future it will optionally contain more information about the user from their account provider, e.g. their email address. Also potentially information that could be used to generate a suggested username.